src/pages/workspaces/workspace/analysis/modals/AzureComputeModal.js

```diff
@@ -228,7 +230,8 @@ export const AzureComputeModalBase = ({
             saturnWorkspaceName: workspaceName
           },
           disk
-        })
+        }, persistentDiskExists,
```

Comment:
Can we remove the line `const [currentPersistentDiskDetails] = useState(currentDisk)` on top? There's no setter, so a a bit confused why a prop is redeclared as state, and tracing through 3 variables to find what this is derived from.